x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/entsearch/engine/action/ListEnginesAction.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.entsearch.engine.action;
+
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.action.ValidateActions;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.StatusToXContentObject;
+import org.elasticsearch.core.Nullable;
+import org.elasticsearch.rest.RestStatus;
+import org.elasticsearch.xcontent.ParseField;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xpack.core.action.util.PageParams;
+import org.elasticsearch.xpack.core.action.util.QueryPage;
+import org.elasticsearch.xpack.entsearch.engine.Engine;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+
+public class ListEnginesAction extends ActionType<ListEnginesAction.Response> {
+
+    public static final ListEnginesAction INSTANCE = new ListEnginesAction();
+    public static final String NAME = "cluster:admin/engine/list";
+
+    public ListEnginesAction() {
+        super(NAME, ListEnginesAction.Response::new);
+    }
+
+    // TODO Check if this should implement IndicesRequest.Replaceable
+    public static class Request extends ActionRequest {
+
+        private static final String DEFAULT_QUERY = "*";
+        private final String query;
+        private final PageParams pageParams;
+
+        public Request(StreamInput in) throws IOException {
+            super(in);
+            this.query = Objects.requireNonNullElse(in.readOptionalString(), DEFAULT_QUERY);
+            this.pageParams = new PageParams(in);
+        }
+
+        public Request(@Nullable String query, PageParams pageParams) {
+            this.query = Objects.requireNonNullElse(query, DEFAULT_QUERY);
+            this.pageParams = pageParams;
+        }
+
+        public String query() {
+            return query;
+        }
+
+        public PageParams pageParams() {
+            return pageParams;
+        }
+
+        @Override
+        public ActionRequestValidationException validate() {
+            // Pagination validation is done as part of PageParams constructor
+            ActionRequestValidationException validationException = null;
+            if (Strings.isEmpty(query())) {
```

Comment:
As this will be validated by the actual search query, I'll punt on this - I haven't found a validation for query strings in the codebase. Happy to do that in a follow-up PR if anyone point me to that!

Reply:
That's fine, the validation is done when the query is built on a per-shard basis so we can rely on search failure to be raised if the syntax is not correct.